test/unit/testExecutorCurrentNameReason.js

```diff
@@ -0,0 +1,241 @@
+'use strict';
+
+const initSteps = require('app/core/initSteps');
+const {assert, expect} = require('chai');
+const ExecutorsWrapper = require('app/wrappers/Executors');
+const executorCurrentNameReasonPath = '/executor-current-name-reason/';
+
+describe('ExecutorCurrentNameReason', () => {
+    const steps = initSteps([__dirname + '/../../app/steps/action/', __dirname + '/../../app/steps/ui']);
+
+    describe('getUrl()', () => {
+        it('returns the url with a * param when no index is given', (done) => {
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const url = ExecutorCurrentNameReason.constructor.getUrl();
+
+            expect(url).to.equal(`${executorCurrentNameReasonPath}*`);
+            done();
+        });
+
+        it('returns the url with the index as a param when an index is given', (done) => {
+            const param = 1;
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const url = ExecutorCurrentNameReason.constructor.getUrl(param);
+
+            expect(url).to.equal(executorCurrentNameReasonPath + param);
+            done();
+        });
+    });
+
+    describe('getContextData()', () => {
```

Comment:
We should also have a unit tests for `getContextData()` to check the following 2 scenarios:

```
} else if (req.params && req.params[0] === '*') {
```
and 
```
} else if (startsWith(req.path, path)) {
```